packages/stencil/src/components/datacom-number-input/datacom-number-input.tsx

```diff
@@ -0,0 +1,96 @@
+import { Component, Host, h, Prop, Element, State, Listen } from '@stencil/core';
+import { getSvg } from '../../common/images/icon-provider';
+
+@Component({
+  tag: 'datacom-number-input',
+  styleUrl: 'datacom-number-input.css',
+  shadow: false,
+})
+export class DatacomNumberInput {
+  @Element() host: HTMLInputElement;
+
+  @Prop() readonly? = false;
+  @Prop() required? = false;
+  @Prop() numberTitle: string;
+  @Prop() name: string;
+  @Prop() text: string;
+  @Prop() min: number;
+  @Prop() max: number;
+  @Prop() step: number;
+  @Prop() help: string;
+  @Prop() message: string;
+  @Prop() disabled: false;
+  @Prop() placeholder: string;
+  @State() value?: number | null = null;
```

Comment:
Couple of things about this "value" state variable:
- A dev would probably like to set the initial value for this field if needed. You should look to make this a mutable prop rather than a state variable so that you can achieve that.
- You probably don't need to define null like you do here and create a union type. Instead just leave they as just "number" and by default the value will be 'undefined' if not populated I believe. (Then you can check against an undefined value rather than null.)